bundles/org.openhab.core.thing/src/main/java/org/openhab/core/thing/internal/ThingManagerImpl.java

```diff
@@ -379,8 +413,12 @@ public void run() {
                     final ThingHandlerFactory oldThingHandlerFactory = findThingHandlerFactory(thing.getThingTypeUID());
                     if (oldThingHandlerFactory != null) {
                         ThingHandler thingHandler = thing.getHandler();
-                        unregisterAndDisposeHandler(oldThingHandlerFactory, thing, thingHandler);
-                        waitUntilHandlerUnregistered(thing, 60 * 1000);
+                        if (thingHandler != null) {
+                            unregisterAndDisposeHandler(oldThingHandlerFactory, thing, thingHandler);
+                            waitUntilHandlerUnregistered(thing, 60 * 1000);
+                        } else {
+                            logger.debug("No ThingHandler to dispose for {}", thing.getUID());
```

Comment:
You can derive the type from the UID and I think you want to know which specific handler is null in this case because there can be many instances but only this one is null for a reason:

```java
                    // Remove the ThingHandler, if any
                    final ThingHandlerFactory oldThingHandlerFactory = findThingHandlerFactory(thing.getThingTypeUID());
                    if (oldThingHandlerFactory != null) {
                        ThingHandler thingHandler = thing.getHandler();
                        if (thingHandler != null) {
                            unregisterAndDisposeHandler(oldThingHandlerFactory, thing, thingHandler);
                            waitUntilHandlerUnregistered(thing, 60 * 1000);
                        } else {
                            logger.debug("No ThingHandler to dispose for {}", thing.getUID());
                        }
                    } else {
                        logger.debug("No ThingHandlerFactory available that can handle {}", thing.getThingTypeUID());
                    }

```